src/sentry/tasks/statistical_detectors.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Dict, List
+
+import sentry_sdk
+from django.utils import timezone
+
+from sentry import options
+from sentry.models.project import Project
+from sentry.snuba import functions
+from sentry.snuba.referrer import Referrer
+from sentry.tasks.base import instrumented_task
+
+logger = logging.getLogger("sentry.tasks.statistical_detectors")
+
+
+ITERATOR_CHUNK = 1_000
+
+
+@instrumented_task(
+    name="sentry.tasks.statistical_detectors.run_detection",
+    queue="performance.statistical_detector",
+    max_retries=0,
+)
+def run_detection() -> None:
+    if not options.get("statistical_detectors.enable"):
+        return
+
+    now = timezone.now()
+
+    performance_projects: List[int] = options.get(
+        "statistical_detectors.enable.projects.performance"
+    )
+    profiling_projects: List[int] = options.get("statistical_detectors.enable.projects.profiling")
+
+    """ disabled for now so we can run experiements
+    for project in RangeQuerySetWrapper(
+        Project.objects.filter(status=ObjectStatus.ACTIVE),
+        result_value_getter=lambda item: item.id,
+        step=ITERATOR_CHUNK,
+    ):
+        if project.flags.has_transactions:
+            performance_projects.append(project.id)
+
+            if len(performance_projects) >= ITERATOR_CHUNK:
+                detect_regressed_transactions.delay(performance_projects)
+                performance_projects = []
+
+        if project.flags.has_profiles:
+            profiling_projects.append(project.id)
+
+            if len(profiling_projects) >= ITERATOR_CHUNK:
+                detect_regressed_functions.delay(profiling_projects, now)
+                profiling_projects = []
+    """
+
+    # make sure to dispatch a task to handle the remaining projects
+    if performance_projects:
+        detect_regressed_transactions.delay(performance_projects)
+        performance_projects = []
+    if profiling_projects:
+        detect_regressed_functions.delay(profiling_projects, now)
+        profiling_projects = []
+
+
+@instrumented_task(
+    name="sentry.tasks.statistical_detectors._detect_regressed_transactions",
+    queue="performance.statistical_detector",
+    max_retries=0,
+)
+def detect_regressed_transactions(project_ids: List[int], **kwargs) -> None:
+    for project_id in project_ids:
+        if not options.get("statistical_detectors.enable"):
```

Comment:
Does the option read need to happen within the loop? Seems like something that could be done once.

Reply:
No, it does not. I was thinking having it inside the loop means we can set it at run time and have it stop the task on the next project id but options are cached anyways so that likely won't work the way I imagined. Additionally, since we'll be adding a new worker pool, the better way will probably be to scale down the worker pool instead.